Dell/recovery_common.py

```diff
@@ -635,6 +637,21 @@ def parse(fname):
 
     return to_install
 
+def create_grub_entries(rec_type='hdd'):
+    '''Create GRUB entry for dell-recovery during ubiquity installation'''
+    env = os.environ
+    rpart = find_factory_partition_stats()
+    #create the grub entry only when both recovery partition and /cdrom/.disk/info exist
+    if rpart and os.path.exists('/cdrom/.disk/info'):
+        rec_text = 'Restore OS to factory state'
+        process_conf_file(original = '/usr/share/dell/grub/99_dell_recovery', \
+                          new = '/target/etc/grub.d/99_dell_recovery',        \
+                          uuid = str(rpart["uuid"]),                          \
+                          number = str(rpart["number"]),                      \
+                          recovery_text = rec_text,
+                          recovery_type = rec_type)
+        os.chmod('/target/etc/grub.d/99_dell_recovery', 0o755)
```

Comment:
With this being moved to this portion of the install I'd like to see target become a configurable parameter for the function.  The reason being is that presumably userspace could also now recover the file in `/etc/grub.d/99_dell_recovery` if it was destroyed.

Reply:
> The reason being is that presumably userspace could also now recover the file in /etc/grub.d/99_dell_recovery if it was destroyed.

I don't quite sure if I understand this.  Could you kindly elaborate more details about this rationale?

Reply:
I don't know for sure that it will be implemented, but potentially there can be a check if recovery command line tool you created is called and this file is missing (such as a user deleted it).  The tool could recreate it again.

Reply:
Got your point.  I will make the change accordingly.